### PR TITLE
Add cancel an asset transfer endpoint

### PR DESCRIPTION
**Pull request by subarashii-huch09**
# Changes made in this PR:
- Add `cancel an asset transfer` endpoint
- Add new status `cancelled`
- Update asset transfer lifecycle diagram and add the diagram in `/public` directory
asset-lifecycle

The authorization `asset-transfers:create` is already included in auth.md & auth.mdx

API design is [here](https://www.notion.so/centrapay/Cancel-An-Asset-Transfer-02edea268e9145fa97dc1efcc7d27767?pvs=4)

### CODE REVIEW

**Comment by Neeraj-Ghodla** [comment]
@subarashii-huch09 You should also update the [Asset Transfer Lifecycle Diagram](https://docs.centrapay.com/api/asset-transfers#asset-transfer-lifecycle) with the cancelled status.

File: legacy/src/api/assets/asset-transfers.md [hunk 1]
```diff
@@ -281,3 +282,37 @@ Returns a [paginated][] list of Asset Transfers.
...
+  "suppressNotifications": false
+}
+{% endjson %}
+
+{% h4 Error Responses %}
+
+| Status |             Code              |                               Description                               |
+| :----- | :---------------------------- | :---------------------------------------------------------------------- |
+| 403    | FORBIDDEN                     | The asset transfer has not been found or there are missing permissions. |
+| 403    | INVALID_ASSET_TRANSFER_STATUS | The asset transfer status is invalid and cannot be cancelled.           |
```
Comment (Neeraj-Ghodla) [inline legacy/src/api/assets/asset-transfers.md:318]: I think a better description here would be

"The asset transfer is in a final status and cannot be canceled."

**Comment by subarashii-huch09** [comment]
- [ ] Update the [Asset Transfer Lifecycle Diagram](https://docs.centrapay.com/api/asset-transfers#asset-transfer-lifecycle) with the cancelled status.

**Review by Neeraj-Ghodla — CHANGES_REQUESTED** [review]

File: legacy/src/api/assets/asset-transfers.md [hunk 2]
```diff
@@ -281,3 +282,37 @@ Returns a [paginated][] list of Asset Transfers.
...
+  "updatedAt": "2020-05-02T01:03:37.222Z",
+  "suppressNotifications": false
+}
+{% endjson %}
+
+{% h4 Error Responses %}
+
+| Status |             Code              |                               Description                               |
+| :----- | :---------------------------- | :---------------------------------------------------------------------- |
+| 403    | FORBIDDEN                     | The asset transfer has not been found or there are missing permissions. |
```
Comment (Ray2126) [inline legacy/src/api/assets/asset-transfers.md:317]: **Change**

We don't need to document generic forbiddens

File: legacy/src/api/assets/asset-transfers.md (hunk 1 above)
Comment (Ray2126) [reply legacy/src/api/assets/asset-transfers.md:318]: @Neeraj-Ghodla Expired asset transfers cannot be cancelled. Expired is not a final status so this isn't necessarily true.

**Comment by Neeraj-Ghodla** [comment]
> # Task
> * [ ]  Update the [Asset Transfer Lifecycle Diagram](https://docs.centrapay.com/api/asset-transfers#asset-transfer-lifecycle) with the cancelled status.
> 
> @Neeraj-Ghodla @Ray2126 @seliwe
> 
> This is the draft diagram with `cancelled` status. Does it look correct? 🙂
> 
> Screenshot 2023-10-26 at 2 02 01 PM

Looks fine to me.

**Comment by MeganSteenkamp** [comment]
@subarashii-huch09 can we please update the new documentation and not the legacy? ❤️
There should be a new file there ready for you. Sorry about the late notice ⭐️

**Comment by subarashii-huch09** [comment]
> @subarashii-huch09 can we please update the new documentation and not the legacy? ❤️ There should be a new file there ready for you. Sorry about the late notice ⭐️

 @MeganSteenkamp  😃 yep, I'll ensure only making updates on the new documentation. Thank you ! 🙌 
 
- [ ]  Update the new documenation, not the legacy

**Comment by subarashii-huch09** [comment]
@Neeraj-Ghodla @MeganSteenkamp @seliwe @Ray2126 
Thank you all for the feedback, the PR is ready for another round of review 😃 🙏

**Review by Neeraj-Ghodla — APPROVED** [review]

**Comment by MeganSteenkamp** [comment]
@subarashii-huch09 if you want to preview your changes you can run the `Deploy Dev` workflow with the `master` branch and then go to https://docs-preview.cp42.click :)

**Comment by subarashii-huch09** [comment]
> @subarashii-huch09 if you want to preview your changes you can run the `Deploy Dev` workflow with the `master` branch and then go to https://docs-preview.cp42.click :)

@MeganSteenkamp  That's really cool ! Thank you for the information!  🧡 🤩

**Pull request merged**